### ISSUE TITLE
⚰ Duplicated MakeMove crashes JVM judge

### ISSUE THREAD

**Issue by Terkwood** [Terkwood/BUGOUT#228]
# Problem statement

`MakeMove` commands intended for redis-streams/KataGo can be duplicated into kafka and cause the JVM judge component to trash.  After that, multiplayer moves end up not being delivered for _anyone_.

judge

```text
⚖️           ️9387d013 ACCEPT   WHITE @ Coord(x=2, y=2) capturing 
📢          09fe07ae MOVE     WHITE Coord(x=6, y=3)
Exception in thread "bugout-judge-26fe66ec-5226-4c6d-bdb8-3697465583ee-StreamThread-1" java.lang.IllegalArgumentException: Parameter specified as non-null is null: method Judge$process$valueJoiner$1.apply, parameter rightValue
	at Judge$process$valueJoiner$1.apply(Judge.kt)
	at Judge$process$valueJoiner$1.apply(Judge.kt:18)
```

at 

```kotlin
        val valueJoiner: ValueJoiner<MakeMoveCmd, GameState, MoveCommandGameState> =
            ValueJoiner { leftValue:
                          MakeMoveCmd,
                          rightValue:
                          GameState ->
                MoveCommandGameState(leftValue, rightValue)
            }
```
## Solution

Duplex was a mistake.

Pick one to close this ticket:

- alter `gateway` so that it responsibly links `(Session_id, BackendCommand) ` to a choice in `Backend`.  (#253) 
-  use an inner join on `game_states_changelog`  stream (as guard) in judge prior to materializing game state... And raise an issue to do the above design change in gateway 

## Merge this reversion

#249 rolls back an unnecessary change set #243 produced as part of this investigation. 

- [x] Revert #249

**Comment by Terkwood** [comment]
kafka box is not responding somehow, so some reboots tend to fix this. let's delete kafka apps

**Comment by Terkwood** [comment]
double check to make sure it's not a problem in gateway box.  try ONLY rebooting kafka box, not gateway box

**Comment by Terkwood** [comment]
Tried to fix this with #240 , #241  -- but to no avail.

**Comment by Terkwood** [comment]
Still seeing this error, _mid-game_, despite the attempted fix in #242.

```text
⚖️           ️777bcbb6 ACCEPT   WHITE @ Coord(x=14, y=2) capturing
📢          777bcbb6 MOVE     BLACK Coord(x=7, y=6)
⚖️           ️777bcbb6 ACCEPT   BLACK @ Coord(x=7, y=6) capturing
📢          777bcbb6 MOVE     WHITE Coord(x=15, y=6)
⚖️           ️777bcbb6 ACCEPT   WHITE @ Coord(x=15, y=6) capturing
📢          a7eb1796 MOVE     WHITE Coord(x=3, y=4)
Exception in thread "bugout-judge-b445b940-190e-4ad7-ab89-79982427323e-StreamThread-1"
java.lang.IllegalArgumentException: Parameter specified as non-null is null: method Jud
ge$process$valueJoiner$1.apply, parameter rightValue
        at Judge$process$valueJoiner$1.apply(Judge.kt)
        at Judge$process$valueJoiner$1.apply(Judge.kt:18)
```
